lightning/src/ln/channel.rs

```diff
@@ -1785,8 +1785,22 @@ impl<Signer: Sign> Channel<Signer> {
 	/// corner case properly.
 	pub fn get_inbound_outbound_available_balance_msat(&self) -> (u64, u64) {
```

Comment:
I think this function name and its usage in `list_channels_with_filter` are confusing balance/liquidity/capacity. IMHO, you have :
- capacity : the size of the funding output value as seen on-chain, already mentioned in this sense by our API in `ChannelInfo`
- liquidity : the in-flight, directed, available value to transfer HTLC, as defined by channel policy (`htlc_maximum_msat` or probabilistic assumptions), already mentioned through `get_route`
- balance : the `to_local` output claimable by the holder without competition, assuming we're not broadcasting revoked state

So as a suggestion i would suggest to recall this function `get_inbound_outbound_available_liquidity_msat` though here we have not providing information on the `max_accepted_htlc`/`max_htlc_value_in_flight` which could effectively restrain the _effective_ liquidity lawfully usable for forwarding/receiving ?

Maybe I would also suggest to drop the "not including pending HTLCs" are in fact they're subtracted to reflect that you can't reuse liquidity occupied by `get_inbound_pending_htlc_stats` to receive more ?

And last point, maybe precise yielded values should be consider "rough" and not considered to apply strict, additional policy enforcement ? E.g close channel if ratio inbound/outbound reaches X ?

Reply:
The docs already indicate that these values are approximate and not exact, so we agree there :). Can we rename an existing function in a later PR? 0.0.99 is already late enough.